droid-hal-device.inc

```diff
@@ -385,6 +389,25 @@ if (grep -q '^TARGET_ARCH := arm64' %{android_root}/device/*/*/BoardConfig*.mk);
 fi
 %endif
 
+%if 0%{?with_system_keys:1}
+KERNEL_DIRS=$(find %android_root -maxdepth 1 -type d \( -name "kernel-*.*" -o -name "kernel" \) &&
+              find %android_root -maxdepth 2 -type d -path "*/linux/kernel")
+
+# Copy trusted keys into kernel source directory
+echo "$KERNEL_DIRS" |
+while IFS= read -r kernel; do
+    cp -r /etc/keys/kernel/* ${kernel}/
```

Comment:
This looks weird how could or should come these keys from the rootfs of the build host?

Reply:
They are supposed to be provided by `system-keys-kernel` package.

Reply:
Shouldn't they go then into /usr/lib?

Reply:
`/usr/lib` is [intended](https://refspecs.linuxfoundation.org/FHS_3.0/fhs/ch04s06.html) for storing object files and libraries. But keys are neither of them, they are much closer to configuration files (which [should go](https://refspecs.linuxfoundation.org/FHS_3.0/fhs/ch03s07.html) into `/etc/*`). Also, Linux Kernel is using /etc/keys/ path [by default](https://github.com/torvalds/linux/blob/122fa8c588316aacafe7e5a393bb3e875eaf5b25/security/integrity/ima/Kconfig#L308) for storing IMA certificate.

So, `/etc/keys` seems to be a better fit here.

Reply:
While I am agree on moving static data to `/usr` (which especially makes sense on RO filesystems), I'm not really sure that `/usr/lib` fits well for storing keys and certificates.

Maybe `/usr/share` then? As `/usr/lib` is primarily targeted for storing executable things (e.g. libraries and plugins), and quite rarely used for storing arbitrary data.